### PR TITLE
🐛 Fix layers view querying by type

### PR DESCRIPTION
**Pull request by aldebaran30701**
### Description of the Change
This PR adds bi-functions to the registry which will allow comparison between type objects and the strings it can be compared with.

Also included in this PR is the creation of a dialog box when the user tries to enable a layer with an invalid query. This is in place of throwing an exception to the user. Hopefully this will be less scary and viewed as user error to prompt review of the query passed in.

### Alternate Designs
Adding red text to the UI of Layers View. This may be a future task as it seemed to want to not refresh the UI at the correct time.

### Why Should This Be In Core?
Adds extra query ability to layers view

### Benefits
This will increase the usability of the query interface in layers view.

### Possible Drawbacks
Possibility of popup appearing on a different display to the main user window.

### Verification Process
Ran queries of type attributes and tried to query every attribute type.

### Applicable Issues
#1002

### CODE REVIEW

**Comment by aldebaran30701** [comment]
Merge this before #1115

File: CoreGraphFramework/src/au/gov/asd/tac/constellation/graph/value/expression/ExpressionCompiler.java [hunk 1]
```diff
@@ -100,7 +116,9 @@ public static Object compileSequenceExpression(SequenceExpression expression, Va
                 final String operatorName = CONVERTER_CLASSES.get(operator.getOperator());
                 final Object result = operators.getRegistry(operatorName).apply(right);
                 if (result == null) {
-                    throw new ExpressionException("Unable to find implementation of operator: " + operatorName + " (" + right.getClass().getCanonicalName() + ")");
+                    Platform.runLater(() -> NotifyDisplayer.displayAlert(QUERY_ERROR, OPERATOR_NOT_FOUND, 
+                            String.format(OPERATOR_ERROR_MSG, operatorName), Alert.AlertType.ERROR));
+                    LOGGER.log(Level.WARNING, OPERATOR_NOT_FOUND_MSG, operatorName);
```
Comment (arcturus2) [inline CoreGraphFramework/src/au/gov/asd/tac/constellation/graph/value/expression/ExpressionCompiler.java:121]: This is a cool way of doing the alert and logging the exception. I would like to do this across the board. :+1:

File: CoreGraphFramework/src/au/gov/asd/tac/constellation/graph/value/operations/Contains.java [hunk 2]
```diff
@@ -43,6 +47,33 @@ public boolean execute(String p1, String p2) {
     public static void register(Operators operators) {
         final OperatorRegistry registry = operators.getRegistry(NAME);
         STRING_OPERATION.register(registry);
+
+        // Used when query is as follows: x contains '1'
```
Comment (arcturus2) [inline CoreGraphFramework/src/au/gov/asd/tac/constellation/graph/value/operations/Contains.java:51]: This is a minor one but I would find it a bit easier to read if you start with "e.g" `// e.g. Used when ...` as it took me a little while to understand that you were providing examples here.

**Comment by aldebaran30701** [comment]
I'm thinking this PR can be closed, and not merged. 
The PR #1115 Includes all changes here, and amendments to it also.

**Comment by arcturus2** [comment]
Closing PR as requested.

**Pull request closed without merging**